lib/prelude/src/wrapper.rs

```diff
@@ -0,0 +1,113 @@
+//! This type defines Wrap / Unwrap utilities. Unwrap is like `Deref` but does not implement
+//! `impl<'a, T> Unwrap for &'a T` in order to make it less error prone. `Wrap` is like `pure` in
+//! applicative functors – if lifts a value to the specific type.
+
+use crate::std_reexports::*;
+
+// ===============
+// === Wrapper ===
+// ===============
+
+/// Trait for any type which wraps other type. See docs of `Wrapper` to learn more.
+pub trait HasContent {
```

Comment:
`KnownContent` will better fit the other similar traits.

Reply:
See the explanation above. `HasContent` means that this struct has a data which can be described as "content". `KnownContent` means that based on this type, we can infer a type of "content". `KnownX` is type-level only utility. Often `HasX` uses `KnownX` as dependency.